### ISSUE TITLE
Installation from packages does not use any configuration after first converge.

### ISSUE THREAD

**Issue by gavinheavyside** [chef-cookbooks/docker#798]
### Cookbook version
2.13.9

### Chef-client version
12.17.44

### Platform Details
Ubuntu trusty, in test kitchen, https://atlas.hashicorp.com/ubuntu/trusty64

### Scenario:
When installing from packages, the config files are written _after_ the service has already started. There is now no restart of the service, due to https://github.com/chef-cookbooks/docker/commit/feffc9ba0fab52377ac208ce1c504d89c8945626, as mentioned in #793.

The result, is that when converging a new node, the docker service does not pick up any configuration, it runs with the package default. In my case, this means that the log driver is wrong, among other things.

After a manual restart of the docker service, the config is picked up correctly.

### Steps to Reproduce:
A simple recipe can illustrate this:

```ruby
include_recipe "chef-apt-docker"

docker_service "default" do
  version        "1.12.5"
  install_method "package
  log_driver     "awslogs"
  action         [:create, :start]
end
```

After converging, `docker info` shows `Logging Driver: json-file`. Restarting the docker service leads to `Logging Driver: awslogs`

### Expected Result:
After converging for the first time, the docker engine should be using the configuration in the `docker_service` block.

### Actual Result:
After converging for the first time, the docker engine is using default settings because it was started before the config was written, and the service is not restarted.

**Comment by someara** [comment]
@lamont-granquist any ideas on how to handle this?

Previously, we were able to notify :restart to new_resource... I can't seem to figure out what to do now that the run_context scopes are broken

-s

**Comment by someara** [comment]
@tas50 any insights to what needs to be incanted here?

How do I simulate the "classic use_inline_resources" behavior

-s

**Comment by smgt** [comment]
I'm guessing changes in chef, why can't you use notify restart?

**Comment by kkeane** [comment]
I have a different symptom but believe that the underlying cause is the same:

A change to attributes of the docker_source will also not cause the systemctl daemon-reload/systemctl restart docker-xxx sequence. This happens on CentOS 7.

**Comment by someara** [comment]
@smgt notifications now "bubble up" into outer run_contexts... 
I'm not sure how to schedule it for the end of the resource implementation.

**Comment by someara** [comment]
I've added an :immediate timing to the original notification and released it in 2.13.11.
It doesn't work quite like it used to, but it's better than nothing for now.

-s
